frontend/components/LSP7Panel.tsx

```diff
@@ -95,10 +105,18 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
         signer
       );
       const lsp7 = tokenContract.connect(signer);
+      let balanceInBaseUnits = '0'
+      if (tokenData && tokenData.decimals !== undefined && rawTokenAmount) {
+        balanceInBaseUnits = ethers.utils.formatUnits(rawTokenAmount, tokenData?.decimals);
```

Comment:
if we never touch the raw value that we receive in tokenUtils as the balance is this step necessary?